website/www/site/content/en/blog/gsoc-25-ml-connectors.md

```diff
@@ -0,0 +1,254 @@
+integrations with mental chatter in the form of fingers crossed, hoping that
+there will be no divergence on the design. Now I can engage people I am working
+with through design docs, making sure my work aligns with their vision, which
+potentially leads to faster PR merges
+- **Skillset:** It was one year before contributing to Beam where I wrote
+professionally in Python, so it was a great opprtunity to brush up on my Python
+skills and seeing how some design patterns are used in practice, like the query
+builder pattern seen in CloudSQL Vector Ingestion in the RAG package. I also
+learned about vector databases and feature stores, and also some AI
+integrations. I also think I got a bit better than before in root cause analysis
+and filtering signals from noise in long log files like PostCommit Python
+workflows
+- **Toolset:** Learning about Beam Python SDK, Milvus, Tecton, Google CloudSQL,
+OpenAI and Anthropic text embedding generators, and lnav for effective log file
+navigation, including their capabilities and limitations
+
+## Tips for Future Contributors
+
+If I have to boil them down to three, they would be:
+
+- **Observing:** Observing how experienced developers in the Beam dev team
+work—how their PRs look, how they write design docs, what kind of feedback they
+get on their design docs and PRs, and how you can apply it (if feasible) to
+avoid getting the same feedback again. What kind of follow-up PRs do they create
+after their initial ones? How do they document and illustrate their work? What
+kind of comments do they post when reviewing other people's related work? Over
+time, you build your own mental model and knowledge base on how the ideal
+contribution looks in this area. There is a lot to learn and explore in an
+exciting, not intimidating way
+- **Orienting:** Understanding your place in the ecosystem and aligning your
+work with the project's context. This means grasping how your contribution fits
+into Beam's architecture and roadmap, identifying your role in addressing
+current gaps, and mapping stakeholders who will review, use, and maintain your
+work. Most importantly, align with both your mentor's vision and the community's
+vision to ensure your work serves the broader goals
+- **Acting:** Acting on feedback from code reviews, design document discussions,
+and community input. This means thoughtfully addressing suggested changes in a
+way that moves the discussion forward, addressing concerns raised by
+maintainers, and iterating on your work based on community guidance. Being
+responsive to feedback, asking clarifying questions when needed, and
+demonstrating that you're incorporating the community's input into your
+contributions given that it is aligned with the project direction
+
+## Who Do I Want To Thank for Making This Journey Possible?
+
+If I have to boil them down to three, they would be:
+
+- **My Mentor, Danny McCormick:** I wouldn't hesitate to say that Danny is the
```

Comment:
Thank you - this is very kind :) It has been awesome working with you as well, I couldn't have asked for a better mentee